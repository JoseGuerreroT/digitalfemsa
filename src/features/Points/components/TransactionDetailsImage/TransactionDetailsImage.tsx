import React from 'react';
import {View, Image, StyleSheet} from 'react-native';

const DEFAULT_IMAGE = require('../../../../assets/images/placerholder-trasaction-item.png');

type TransactionDetailsImageProps = {
  image?: string;
};

const TransactionDetailsImage: React.FC<TransactionDetailsImageProps> = ({
  image,
}) => (
  <View style={styles.container}>
    <Image source={image ? {uri: image} : DEFAULT_IMAGE} style={styles.image} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    padding: 75,
    backgroundColor: '#FFF',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,
  },
  image: {
    width: 200,
    height: 200,
  },
});

export default TransactionDetailsImage;
