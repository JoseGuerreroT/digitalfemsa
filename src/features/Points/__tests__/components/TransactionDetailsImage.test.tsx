import React from 'react';
import {render} from '@testing-library/react-native';

import TransactionDetailsImage from '../../components/TransactionDetailsImage';

describe('TransactionDetailsImage', () => {
  test('01 - renders correctly', () => {
    const {toJSON} = render(<TransactionDetailsImage />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('02 - displays the default image if no image is provided', () => {
    const DEFAULT_IMAGE =
      '../../../src/assets/images/placerholder-trasaction-item.png';

    const {getByLabelText} = render(<TransactionDetailsImage />);
    const image = getByLabelText('Transaction Image');
    expect(image.props.source.testUri).toBe(DEFAULT_IMAGE);
  });

  test('displays the provided image if an image URL is provided', () => {
    const imageUrl = 'https://example.com/image.jpg';
    const {getByLabelText} = render(
      <TransactionDetailsImage image={imageUrl} />,
    );
    const image = getByLabelText('Transaction Image');
    expect(image.props.source.uri).toBe(imageUrl);
  });
});
