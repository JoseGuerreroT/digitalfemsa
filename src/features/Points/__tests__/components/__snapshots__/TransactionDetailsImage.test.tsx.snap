// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetailsImage 01 - renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 10,
      "justifyContent": "center",
      "padding": 75,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 4,
    }
  }
>
  <Image
    accessibilityLabel="Transaction Image"
    role="img"
    source={
      {
        "testUri": "../../../src/assets/images/placerholder-trasaction-item.png",
      }
    }
    style={
      {
        "height": 200,
        "width": 200,
      }
    }
  />
</View>
`;
